injected/src/features/element-hiding.js

```diff
@@ -360,19 +361,14 @@ export default class ElementHiding extends ContentFeature {
         } else {
             applyRules(activeRules);
         }
-        // single page applications don't have a DOMContentLoaded event on navigations, so
-        // we use proxy/reflect on history.pushState to call applyRules on page navigations
-        const historyMethodProxy = new DDGProxy(this, History.prototype, 'pushState', {
-            apply(target, thisArg, args) {
-                applyRules(activeRules);
```

Comment:
@dharb  This wasn't applying after the URL had changed, which meant it might not have worked correctly in all cases. This change shall make the apply rules code run after the url bar has changed.

Reply:
Also to note I've added support for the navigation api when it's available; this should be less risk on Chromium based browsers as we'll not need to do any wrapping.